ic-os/defs.bzl

```diff
@@ -44,6 +44,25 @@ def icos_build(
 
     image_deps = image_deps_func(mode, malicious)
 
+    # -------------------- Pre-check --------------------
+    # Verify that all the referenced components exist
+    native.genrule(
+        name = name + "_pre_check",
+        srcs = [k for k, v in image_deps["component_files"].items()],
+        outs = [name + "_pre_check_result.txt"],
+        cmd = """
+        echo "Running pre_check for {name}" > $@
+        for file in $(SRCS); do
```

Comment:
I believe srcs is just the list of file sources. We still need to check if they exist and fail accordingly.

I tested the old version with a non-existant file in guestos.bzl and it ran without error.

Reply:
I had also thought the dependencies should evaluate first, but maybe it is lazy in some cases?

We may also want to include in the final target, so that it is checked when running `bazel build //ic-os/guestos/envs/dev` for example. Leaving it not tagged as `manual` means it will already be built by `//...`, which is good.

Reply:
Conclusion: https://dfinity.slack.com/archives/C02713U2LMN/p1723662707067249?thread_ts=1723662704.048699&cid=C02713U2LMN